### ISSUE TITLE
Moving a local object in a return statement prevents copy elision

### ISSUE THREAD

**Issue by ned14** [martinus/robin-hood-hashing#89]
```
../thirdparty/llfio/include/../../../src/core/../robin_hood.hpp:1259:33: warning: moving a local object in a return statement prevents copy elision [-Wpessimizing-move]
 1259 |             return std::move(tmp);
      |                                 ^
../thirdparty/llfio/include/../../../src/core/../robin_hood.hpp:1259:33: note: remove ‘std::move’ call
```

**Comment by martinus** [comment]
what compiler are you using? I wonder why my g++10.1.0 doesn't detect this when I add that flag

**Comment by martinus** [comment]
Ah, it works because I didn't have a test for it :man_facepalming: 

Thanks for finding, fixed in 8eb44ddef7118f1045781895108cdd4d6cbbd0e2
